Topics/Development_Process/Azure_Functions.md

```diff
@@ -0,0 +1,193 @@
+
+## Create an Azure Function
+
+Open up a terminal and `cd` into a directory you'd like to create our Azure Function project. Once there, run `func init`. Select `dotnet` followed by `c#`, and your Azure Functions project will be initialized.
+
+We will be creating an Azure Function that responds to HTTP triggers. If you installed the Azure Functions extension in VS Code:
+
+1. Click the Azure extension button
+2. Hover over "WORKSPACE" and select the Azure Function button
+3. Select Create Function
+4. Select HTTP Trigger template
+5. Name your Function `funcHttp`
+6. Provide a namespace `Example.Demo`
+7. Select Anonymous for access rights
+
+Alternatively, I have provided the generated example below. Create a `funcHttp.cs` file inside the top-level directory of the project, and add the following:
+
+```cs
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Example.Demo
+{
+    public static class funcHttp
+    {
+        [FunctionName("funcHttp")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            log.LogInformation("C# HTTP trigger function processed a request.");
+
+            string name = req.Query["name"];
+
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+            dynamic data = JsonConvert.DeserializeObject(requestBody);
+            name = name ?? data?.name;
+
+            string responseMessage = string.IsNullOrEmpty(name)
+                ? "This HTTP triggered function executed successfully. Pass a name in the query string or in the request body for a personalized response."
+                : $"Hello, {name}. This HTTP triggered function executed successfully.";
+
+            return new OkObjectResult(responseMessage);
+        }
+    }
+}
+```
+
+Finally, inside a terminal `cd` into the project and run `func start`. This will run all your defined Azure functions locally. In our case, we just have this single function that responds to HTTP requests. To test out your function, you can:
+
+- Open a browser and enter the url http://localhost:7071/api/funcHttp?name=MyName
+- Use a tool like `curl` or Postman to make a POST request to the server with a request body containing a "name".
+
+That's it! You've created your first Azure function.
+
+## Deploy an Azure Function
+
+First, change the `AzureWebJobsStorage` property inside `local.settings.json` to the value of the connection string you copied when creating the Storage account.
+
+Now, open up a terminal and run `az login`. This will open a browser page and prompt you to sign in to your Azure account.
```

Comment:
Maybe obvious to those more familiar, but since I'm not familiar with this.. what directory would one want to be in when running `az login`? In the location of the local.settings.json or the project directory? Or does it not matter?

Reply:
Not at all obvious if you've never used the CLI tool before!

You can run the `az login` command in any directory. You will be authenticated to run subsequent `az` commands in any directory now, such as deploying a resource, creating a new one tied to your subscription, etc.

I have updated the tutorial to make this more clear.